### PR TITLE
Added optional Pre-Renewal Brasilis

### PR DESCRIPTION
**Pull request by Daegaladh**
* **Addressed Issue(s)**: N/A

* **Server Mode**: Pre-renewal

* **Description of Pull Request**: Adds the missing data and NPCs to fully enable the Pre-Renewal version of Brasilis, from official data from bRO and jRO.
Since we follow kRO, this is added as optional and commented out by default.
To enable it just uncomment the files from /npc/pre-re/scripts_athena.conf and /db/import/mob_summon.yml

Special thanks to @Playtester for the spawns info.

jRO's spawns info:;
https://web.archive.org/web/20100507033319/https://ro.silk.to/map/bra_fild01.html
https://web.archive.org/web/20100505135012/http://ro.silk.to/map/bra_dun01.html
https://web.archive.org/web/20100505131755/http://ro.silk.to/map/bra_dun02.html

### CODE REVIEW

File: npc/merchants/inn.txt [hunk 1]
```diff
@@ -203,6 +203,45 @@ ve_in,157,219,5	script	Inn Master#Receptionist	709,{
 	}
 }
 
+//======================== Brasilis =================================
+bra_in01,27,24,3	script	Hotel Keeper#bra1	478,{
+	mes "[Hotel Keeper]";
+	mes "Welcome to the beautiful Brasilis Hotel.";
+	next;
+	switch(select("Save:Rest -5000 zeny")) {
```
Comment (Atemo) [inline npc/merchants/inn.txt:211]: Avoid ":" in select, less readable than coma

File: npc/merchants/inn.txt [hunk 2]
```diff
@@ -203,6 +203,45 @@ ve_in,157,219,5	script	Inn Master#Receptionist	709,{
...
+		case 1:
+			mes "[Hotel Keeper]";
+			mes "Ok then, enjoy your stay.";
+			close;
+		case 2:
+			mes "[Hotel Keeper]";
+			mes "Your respawn has been saved here at the hotel. I hope that you enjoy your stay here in Brasilis.";
+			savepoint "bra_in01",144,69,1,1;
+			close;
+		}
```
Comment (Atemo) [inline npc/merchants/inn.txt:226]:
```suggestion
		}
		end;
```

File: npc/merchants/inn.txt [hunk 3]
```diff
@@ -203,6 +203,45 @@ ve_in,157,219,5	script	Inn Master#Receptionist	709,{
...
+			set Zeny, Zeny-5000;
+			percentheal 100,100;
+			warp "bra_in01",144,69;
+			end;
+		}
+		else {
+			mes "[Hotel Keeper]";
+			mes "I'm sorry, but the service charge is 5,000 zeny per night.";
+			close;
+		}
```
Comment (Atemo) [inline npc/merchants/inn.txt:241]: Use "else" when the script continue after the "if"
```suggestion
		mes "[Hotel Keeper]";
		mes "I'm sorry, but the service charge is 5,000 zeny per night.";
		close;
```

File: npc/pre-re/guides/guides_brasilis.txt [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+			mes "Is there anything else I can do for you?";
+			viewpoint 1,254,248,5,0x55FF33;
+			close;
+		case 5:
+			mes "[Brasilis Guide]";
+			mes "The iconic monument of Brasilis, the Verass Monument stands at ^3355FF+^000000.";
+			mes "Is there anything else I can do for you?";
+			viewpoint 1,195,235,6,0x3355FF;
+			close;
+		}
```
Comment (Atemo) [inline npc/pre-re/guides/guides_brasilis.txt:60]:
```suggestion
		}
		end;
```

File: npc/pre-re/mobs/dungeons/bra_dun.txt [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+//===== Additional Comments: =================================
+//= 1.0 Official spawns from bRO and custom estimation from jRO
+//============================================================
+
+//bRO Pre-Renewal spawns
+
+//==================================================
+// bra_dun01 - Behind the Waterfall
+//==================================================
+bra_dun01,0,0	monster	Iara	2069,60,5000
```
Comment (Atemo) [inline npc/pre-re/mobs/dungeons/bra_dun.txt:20]: "0,0" is optionnal
```suggestion
bra_dun01	monster	Iara	2069,60,5000
```

File: npc/pre-re/scripts_athena.conf [hunk 6]
```diff
@@ -98,3 +98,23 @@ npc: npc/pre-re/quests/quests_nameless.txt
...
+// -------------------- Pre-Renewal Brasilis --------------------
+// Brasilis was introduced in Pre-Renewal only for bRO and
+// jRO official servers with differences in monster stats,
+// spawns and quests.
+// It was added in Renewal for every other official server.
+// Since we follow kRO this is disabled by default.
+// Uncomment the following NPC scripts to enable bRO/jRO's
+// Pre-Renewal version of Brasilis.
+// You also need to uncomment the data from file
+// mob_summon.yml in \db\import\ folder
```
Comment (Lemongrass3110) [inline npc/pre-re/scripts_athena.conf:111]:
```suggestion
// mob_summon.yml in /db/import\ folder
```

File: npc/pre-re/mobs/fields/brasilis.txt [hunk 7]
```diff
@@ -0,0 +1,53 @@
+//===== rAthena Script =======================================
```
Comment (Lemongrass3110) [inline npc/pre-re/mobs/fields/brasilis.txt:1]: Please use the new header

File: npc/pre-re/mobs/dungeons/bra_dun.txt [hunk 8]
```diff
@@ -0,0 +1,68 @@
+//===== rAthena Script =======================================
```
Comment (Lemongrass3110) [inline npc/pre-re/mobs/dungeons/bra_dun.txt:1]: Please use the new header

File: npc/pre-re/guides/guides_brasilis.txt [hunk 9]
```diff
@@ -0,0 +1,81 @@
+//===== rAthena Script =======================================
```
Comment (Lemongrass3110) [inline npc/pre-re/guides/guides_brasilis.txt:1]: Please use the new header

File: npc/pre-re/cities/brasilis.txt [hunk 10]
```diff
@@ -0,0 +1,24 @@
+//===== rAthena Script =======================================
```
Comment (Lemongrass3110) [inline npc/pre-re/cities/brasilis.txt:1]: Please use the new header

File: npc/custom/warper.txt [hunk 11]
```diff
@@ -110,7 +110,16 @@ function Pick {
 	Go(.@map$,.@x,.@y);
 }
 function Restrict {
-	if ((getarg(0) == "RE" && !checkre(0)) || (getarg(0) == "Pre-RE" && checkre(0))) {
+	.@type = getarg(0);
+	
+	if (.@type == "Brasilis") {
+		if (getnpcid(0, "Crewman_bra2"))
```
Comment (Lemongrass3110) [inline npc/custom/warper.txt:116]:
```suggestion
		// If the official warper to Brasilis is enabled, we lift the restriction
		if (getnpcid(0, "Crewman_bra2") != 0)
```

File: npc/custom/warper.txt [hunk 12]
```diff
@@ -110,7 +110,16 @@ function Pick {
...
+			return;
+		else
+			.@type = "RE";
```
Comment (Lemongrass3110) [inline npc/custom/warper.txt:119]:
```suggestion
		// Otherwise we apply the renewal restriction
		.@type = "RE";
```

**Review by Lemongrass3110 — APPROVED** [review]

**Comment by Lemongrass3110** [comment]
PR name and therefore commit name is exceeding 50 characters again, please consider finding a shorter name.
Maybe: Added Pre-Renewal Brasilis
You can put the details into the commit message itself.

File: npc/merchants/inn.txt [hunk 13]
```diff
@@ -203,6 +203,44 @@ ve_in,157,219,5	script	Inn Master#Receptionist	709,{
...
+			savepoint "bra_in01",144,69,1,1;
+			close;
+		}
+		end;
+	case 2:
+		if (Zeny > 4999) {
+			mes "[Hotel Keeper]";
+			mes "I will show you a great room.";
+			close2;
+			set Zeny, Zeny-5000;
```
Comment (Atemo) [inline npc/merchants/inn.txt:233]: This shouldn't happen in theory
```suggestion
			if (Zeny < 5000)
				end;
			Zeny -= 5000;
```

File: npc/merchants/inn.txt [hunk 14]
```diff
@@ -212,7 +250,7 @@ function	script	F_InnMaid	{
 	mes getarg(1) + ".";
 	mes "How may I help you?";
 	next;
-	switch(select("Save:Take a Rest -> 5000 zeny:Cancel")) {
+	switch(select("Save:Take a Rest -> 5000 zeny","Cancel")) {
```
Comment (Atemo) [inline npc/merchants/inn.txt:253]:
```suggestion
	switch(select("Save:Take a Rest -> 5000 zeny:Cancel")) {
```

File: npc/re/scripts_athena.conf [hunk 15]
```diff
@@ -13,8 +13,8 @@ npc: npc/re/airports/izlude.txt
 npc: npc/re/battleground/bg_common.txt
 
 // --------------------------- Cities ---------------------------
+npc: npc/cities/brasilis.txt
```
Comment (Atemo) [inline npc/re/scripts_athena.conf:16]: imo, we should follow the current structure
re files in `npc/re/scripts_athena.conf`
prere files in `npc/prere/scripts_athena.conf`
same content in `npc/scripts_athena.conf`

= brasilis npcs should be duplicated. Then uncommented in `npc/re/scripts_athena.conf`, and commented in `npc/prere/scripts_athena.conf`
or
brasilis npcs should be common to both mode but removed via delnpc in `npc/prere/scripts_athena.conf` by default

**Review by aleos89 — APPROVED** [review]

**Review by Playtester — APPROVED** [review]

**Comment by krad309** [comment]
There's a bug that's causing custom warper to fail. In `function Restrict {` `.@type` should be `.@type$`

**Comment by Daegaladh** [comment]
> There's a bug that's causing custom warper to fail. In `function Restrict {` `.@type` should be `.@type$`

You're right, I'll fix it as soon as I can

**Pull request merged**